src/components/Checkbox/Checkbox.js

```diff
@@ -0,0 +1,30 @@
+import React, { useEffect, useState } from 'react';
+import './Checkbox.scss';
+
+const Checkbox = (props) => {
+  const { name, value, label, onChange, data } = props;
+  const [checked, setChecked] = useState(false);
+
+  useEffect(() => {
+    if (value === data) {
+      setChecked(true);
+    }
+  }, [value, data]);
```

Comment:
checked state를 만들 필요 없이, 받아온 props를 활용하면 될 것 같습니다.
```suggestion
const Checkbox = (props) => {
  const { name, value, label, onChange, data } = props;
  const checked = value === data;
```